File: app/src/main/java/com/my/mvpframe/ui/MainActivity.java
package com.my.mvpframe.ui;

import android.content.Intent;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.my.mvpframe.R;
import com.my.mvpframe.appbase.BaseActivity;
import com.my.mvpframe.db.AppDatabase;
import com.my.mvpframe.db.User;
import com.my.mvpframe.db.UserDao;
import com.my.mvpframe.rxbus.RxBus;
import com.my.mvpframe.rxbus.RxBusConstants;
import com.my.mvpframe.ui.contract.MainContract;
import com.my.mvpframe.ui.presenter.MainPresenter;

import butterknife.BindView;
import butterknife.OnClick;
import io.reactivex.functions.Consumer;

public class MainActivity extends BaseActivity<MainContract.View,MainPresenter> implements MainContract.View {

    @BindView(R.id.textview)
    TextView textview;
    @BindView(R.id.textview1)
    TextView textview1;
    private MainPresenter presenter;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initView() {
        textview.setOnClickListener(view -> {
            startActivity(new Intent(this,DynamicEffectActivity.class));
        });

        RxBus.getIntanceBus().register(RxBusConstants.TYPE_ONE, (Consumer<String>) o -> textview.setText(o));
        RxBus.getIntanceBus().register(RxBusConstants.TYPE_TWO, (Consumer<String>) o -> textview.setText(o));
        RxBus.getIntanceBus().register(RxBusConstants.TYPE_THREE, (Consumer<String>) o -> textview.setText(o));

        //新增的分支dev
    }

    @Override
    protected MainPresenter initPresenter() {
        presenter = new MainPresenter(this);
        return presenter;
    }

    @Override
    public void showErrorTip(String msg) {

    }

    @Override
    public void showTvResult() {

    }

    @Override
    public void showToast(String str) {
        Log.i("123", str);
    }

    @OnClick({R.id.textview1, R.id.textview2, R.id.textview3, R.id.textview4})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.textview1:// 添加第一组数据
                final UserDao userDao = AppDatabase.getInstance(this).userDao();
                final User user = new User(1,"Z","J","18",null);
                presenter.getInsertUser(userDao,user);
                RxBus.getIntanceBus().post(RxBusConstants.TYPE_ONE,"我是"+user.getFirstName().toString());
                break;
            case R.id.textview2:// 添加第二组数据
                UserDao userDao1 = AppDatabase.getInstance(this).userDao();
                User user1 = new User(2,"T","Y","18",null);
                presenter.getInsertUser(userDao1,user1);
                RxBus.getIntanceBus().post(RxBusConstants.TYPE_TWO,"我是"+user1.getFirstName().toString());
                break;
            case R.id.textview3:
                UserDao userDao2 = AppDatabase.getInstance(this).userDao();
                presenter.getAllUser(userDao2);
                RxBus.getIntanceBus().post(RxBusConstants.TYPE_THREE,"我是-------");
                break;
            case R.id.textview4:
                RxBus.getIntanceBus().unRegister(RxBusConstants.TYPE_ONE);

                break;
        }
    }
}

File: app/src/main/java/com/my/mvpframe/ui/MainActivity.kt
package com.my.mvpframe.ui

import android.content.Intent
import android.util.Log
import com.lucky.netlibrary.BasePresenter
import com.lucky.netlibrary.BaseView
import com.my.mvpframe.R
import com.my.mvpframe.appbase.BaseActivity
import com.my.mvpframe.appbase.BaseFragment
import com.my.mvpframe.bean.TabBean
import com.my.mvpframe.ui.contract.MainContract
import com.my.mvpframe.ui.presenter.MainPresenter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity<MainContract.View, MainPresenter>(), MainContract.View {

    private var presenter: MainPresenter? = null
    private var currentFragment: BaseFragment<BaseView,BasePresenter<BaseView>>? = null

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initView() {
        val listOf = listOf(
                TabBean("界面1", R.drawable.unfinish),
                TabBean("界面2", R.drawable.unfinish),
                TabBean("界面3", R.drawable.unfinish),
                TabBean("界面4", R.drawable.unfinish)
        )
        commontab.setTabData(listOf)
//        newAct.setOnClickListener { v -> startActivity(Intent(this, NewEffectActivity::class.java)) }
    }

//    private fun switchFragment() {
//        val fragments = supportFragmentManager.fragments
//        fragments.forEach {
//            supportFragmentManager.beginTransaction()
//                    .hide(it)
//                    .commitAllowingStateLoss()
//        }
//
//        if (currentFragment in fragments) {
//            supportFragmentManager.beginTransaction()
//                    .show(currentFragment)
//                    .commitAllowingStateLoss()
//        } else {
//            supportFragmentManager.beginTransaction()
//                    .add(R.id.flContainer, currentFragment, currentCheckType)
//                    .addToBackStack(null)
//                    .commitAllowingStateLoss()
//        }
//    }

    override fun initPresenter(): MainPresenter? {
        presenter = MainPresenter(this)
        return presenter
    }

    override fun showErrorTip(msg: String) {

    }

    override fun showTvResult() {

    }

    override fun showToast(str: String) {
        Log.i("123", str)
    }
}
